app/services/security/cspHeader.server.ts

```diff
@@ -0,0 +1,54 @@
+import { serverConfig } from "~/config/config.server";
+
+export const getCspHeader = (args: {
+  nonce: string;
+  environment: string;
+  additionalConnectSrc?: string[];
+}) => {
+  const directives: Record<string, string[]> = {
+    "default-src": ["'self'"],
+    "script-src": [
+      "'self'",
+      "https:",
+      `'nonce-${args.nonce}'`,
+      "'strict-dynamic'",
+      "eu-assets.i.posthog.com", // see https://posthog.com/docs/session-replay/troubleshooting#3-content-security-policy
+    ],
+    "frame-src": ["www.youtube-nocookie.com"],
```

Comment:
Let's keep this to self for as long as possible, we don't embed any content from YouTube so far...
```suggestion
    "frame-src": ["'self'"],
```